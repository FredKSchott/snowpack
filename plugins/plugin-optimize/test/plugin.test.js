const path = require('path');
const fs = require('fs');

const plugin = require('../plugin').default;
const {getSnowpackPluginOutputSnapshotSerializer} = require('./serializer');

describe('@snowpack/plugin-optimize', () => {
  beforeEach(() => {
    expect.addSnapshotSerializer(getSnowpackPluginOutputSnapshotSerializer(__dirname));

    const originalWriteFileSync = fs.writeFileSync;
    fs.writeFileSync = jest
      .fn()
      .mockName('fs.writeFileSync')
      .mockImplementation((path, ...args) => {
        if (path.startsWith(__dirname)) return;

        // write files outside of the current folder
        originalWriteFileSync(path, ...args);
      });

    console.log = jest.fn().mockName('console.log');
  });

  it('minimal - no options', async () => {
    const pluginInstance = plugin({
      buildOptions: {},
    });

    await pluginInstance.optimize({
      buildDirectory: path.resolve(__dirname, 'stubs/minimal/'),
    });

    expect(fs.writeFileSync).toMatchSnapshot('fs.writeFileSync');
    expect(console.log).toMatchSnapshot('console.log');
  });

  it('minimal - no minification', async () => {
    const pluginInstance = plugin(
      {
        buildOptions: {},
      },
      {
        minifyJS: false,
        minifyCSS: false,
        minifyHTML: false,
      },
    );

    await pluginInstance.optimize({
      buildDirectory: path.resolve(__dirname, 'stubs/minimal/'),
    });

    expect(fs.writeFileSync).toMatchSnapshot('fs.writeFileSync');
    expect(console.log).toMatchSnapshot('console.log');
  });

  it('no HTML minification, with preloadModules', async () => {
    const pluginInstance = plugin(
      {
        buildOptions: {},
      },
      {
        minifyHTML: false,
        preloadModules: true,
      },
    );

    await pluginInstance.optimize({
      buildDirectory: path.resolve(__dirname, 'stubs/minimal/'),
    });

    expect(fs.writeFileSync).toMatchSnapshot('fs.writeFileSync');
    expect(console.log).toMatchSnapshot('console.log');
  });

  it('minimal - target', async () => {
    const pluginInstance = plugin(
      {
        buildOptions: {},
      },
      {
        target: ['es2018'],
      },
    );

    await pluginInstance.optimize({
      buildDirectory: path.resolve(__dirname, 'stubs/minimal'),
    });

    expect(fs.writeFileSync).toMatchSnapshot('fs.writeFileSync');
    expect(console.log).toMatchSnapshot('console.log');
  });
});
