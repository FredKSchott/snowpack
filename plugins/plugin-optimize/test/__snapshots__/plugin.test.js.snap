// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@snowpack/plugin-optimize minimal - no minification: console.log 1`] = ``;

exports[`@snowpack/plugin-optimize minimal - no minification: fs.writeFileSync 1`] = ``;

exports[`@snowpack/plugin-optimize minimal - no options: console.log 1`] = ``;

exports[`@snowpack/plugin-optimize minimal - no options: fs.writeFileSync 1`] = `
FILE: stubs/minimal/index.html
<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><title>Minimal test</title><link rel="stylesheet" href="style.css"></head><body><script type="module" src="index.js"></script></body></html>

--------------------------------------------------------------------------------

FILE: stubs/minimal/index.js
(()=>{function n(o){console.log(o)}n("test")})();


--------------------------------------------------------------------------------

FILE: stubs/minimal/style.css
body{border:1px solid red}
`;
